src/Badges.sol

```diff
@@ -143,6 +143,28 @@ contract Badges is
     return owner_;
   }
 
+  function deactivateBadge(
+    uint256 _raftTokenId,
+    uint256 _badgeId,
+    string memory _reason
+  ) external {
+    require(specDataHolder.getRaftOwner(_raftTokenId) == msg.sender, "createSpec: unauthorized");
+    // check the deactivation permissions
+    badgeIdToDeactivationReason[_badgeId] = _reason;
+  }
+
+  function reactivateBadge(uint256 _raftTokenId, uint256 _badgeId) external {
```

Comment:
I would suggest renaming this to `reinstateBadge`

however, this is a product question i.e., whether to actually support this or not. Do we already have a confirmation on this requirement?